src/backend/distributed/commands/utility_hook.c

```diff
@@ -720,7 +731,7 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 	 */
 	if (EnableDDLPropagation)
 	{
-		if (ops && ops->postprocess)
+		if (ops && ops->postprocess && shouldExecutePropagationSteps)
```

Comment:
Why do we need this check in Postprocess? If the address is not valid, PG should have already errored out?